쥬스 메이커 [STEP2] Whales

WhalesJin opened the pull request:
안녕하세요 mmim(@joSH0318)🐦!!
머쨍이 밈에게 열심히 밈토링 받고있는 머쨍이가 될 Whales🐋입니다!!
부족한 부분들 채워주셔서 항상 감사드립니다!! 🙇🏻‍♀️
이번 스텝도 잘 부탁드립니다 🙏🏻

### 📝 고민한 점
> 화면 전환 기법
- `재고 수정` 버튼을 끌어다 `재고 추가` 화면과 연결을 했더니 아래와 같은 선택지가 나왔습니다.
    <Img src = "https://github.com/yagom-academy/ios-juice-maker/assets/124643545/1bd5d259-6a4d-4456-8df1-ce07fb7611fb" width="200"/>
- 세그를 고를 때는 `Adaptive Segue`가 현재 환경에 따라 자동적으로 행동을 조정하기 때문에 `Nonadaptive Segue` 보다 가능하면 `Adaptive Segue`를 고르는 것이 좋다고 되어있어서 `Action Segue` 쪽으로 방향을 잡았습니다. (공식문서 - [Using Segues](https://developer.apple.com/library/archive/featuredarticles/ViewControllerPGforiPhoneOS/UsingSegues.html))
- 어떤 차이가 있는지 보고싶어 하나씩 골라서 테스트를 해보았습니다. 
    - Show
        <Img src = "https://github.com/yagom-academy/ios-juice-maker/assets/124643545/cb87f9ac-fc0b-4aa9-bcbf-b968b2db2633" width="600"/>
    - Show Detail
        <Img src = "https://github.com/yagom-academy/ios-juice-maker/assets/124643545/94845f3f-f498-4f72-9aa3-17f7b6827a11" width="600"/>
    - Present Modally
        <Img src = "https://github.com/yagom-academy/ios-juice-maker/assets/124643545/c25866f8-6e7b-4a13-a983-deb605daa6fb" width="600"/>
- 이 중 다시 돌아가는 버튼이 자동으로 생성되는 `Show`가 마음에 들어 더 찾아보니
    - 뷰 컨트롤러들은 특히 `method`를 `override`하고 다른 동작을 구현하는데에 이 `Segue 타입`을 쓴다는 설명을 보고 재고 추가하는 `method`가 들어갈 다음 뷰컨트롤러에 적합할 것 같아서 `Show`로 골랐습니다.
![](https://hackmd.io/_uploads/SktQZfBHn.png)

> `Alert` 띄우기
- `UIAlertController` 키워드로 `Alert` 띄우는 건 성공했지만 닫기 버튼이 없어서 닫을 수가 없었습니다.
- `Action`을 따로 만들어 넣어줘야는 걸 배우고 `UIAlertAction` 형태의 `Action`을 만들어 넣어줬더니 닫기 버튼이 활성화되었고 같은 형태로 재고 부족 시의 `Alert` 창도 만들었습니다.
- `Action`의 `Style`에 대한 고민을 했는데 `Alert` 창을 닫는 기능을 수행하는 `닫기`나 `아니오`의 경우는 `default`스타일로, 뷰가 바뀌는 기능을 수행해야하는 `예` 버튼은 `destructive` 스타일로 구현햇습니다.
- [AlertAction.Style](https://developer.apple.com/documentation/uikit/uialertaction/style)
  <Img src = "https://github.com/yagom-academy/ios-juice-maker/assets/124643545/cf05e83f-b151-4394-ba48-06a13237ad23" width="500"/>

> 스레드 오류 - `unrecognized selector sent to instance` 
- 버튼을 눌렀을 때 발생한 셀렉터 인스턴스가 함수에 제대로 전달되지 않았을 때 발생한 오류
1. 확인해보니 함수명 오타 수정으로 인해 연결이 제대로 되지 않은 것을 확인하여 수정했습니다.
2. 후에 한 번 더 같은 오류가 떴는데 분명 여기저기 연결을 다 확인했음에도 불구하고 계속 오류가 떠서 한참을 애먹었습니다.
    - `throws` 키워드가 들어간 에러 던지는 함수였는데 그 전에 코드를 작성하다가 `try`나 `throws`를 썼다가 생겼던 오류가 생각나서 `throws` 키워드를 삭제해보았습니다.
    - 원래 있던 오류가 사라졌으나 왜 `throws` 키워드에 저런 오류 문구가 뜬건지에 대한 의문은 여전히 해결하지 못하였습니다.

> 재고량 띄우기
- 그때 그때 바뀐 재고량만 보여주는게 효율적일거라 생각해서 쥬스별로 주문 할 때마다 해당하는 과일의 재고만 바꿔주도록 구현하였습니다.
- 그 편 효율적으로 보이긴 하지만 혹시나 생길 오류들(예를 들어 버튼이 두 개가 동시에 눌렸다면? 다 같이 바뀌어야 안전)을 반영하기 쉽지않아 **레이블 정보와 내 재고가 같을지 확신할 수 없다**는 조언을 받았습니다.
- 그러한 부분을 고려하면 하나씩 보여주는 것보다 `전체를 다시 보여주는 편`이 더 안전할 수 있다고 생각되어 수정하였습니다.

> 은닉화
- 재고량을 보기 위해서 은닉화 되어 있는 과일 인벤토리의 `private`를 삭제하였습니다.
- '갖다쓰기 편하려고 본넷을 열어버렸다.'라는 표현을 듣고서야 이 은닉화를 쉽게 본 것에 대한 문제점과 경각심을 느꼈습니다.  
- 프로퍼티를 은닉화 하고 함수로 호출하는 방법으로 해결하였습니다. 
<br>

### 🙏🏻 조언을 구하는 부분
> 함수를 통합하는 더 좋은 방법
- 여러 개의 반복되는 함수를 하나로 합치는 과정을 오래 고민했습니다.
- 검색해보니 `tag` 키워드를 쓰는 방법을 찾았는데 그렇게 되면 숫자 입력이 들어가야하는데 쥬스주문에는 정해진 순서나 명확한 숫자가 정해지지 않는다고 생각해서 채택하지 않았습니다. ([참고 블로그](https://g1embed.tistory.com/8))
- 버튼 이름에 쥬스 타입 케이스 이름과 같은 단어가 들어있어서 그걸 가져다 쓰면 좋겠다라는 생각을 했는데 결국 방법을 찾지 못했습니다.
- 가장 현실적인 방법으로 switch 문을 선택하였는데 약간의 무식한 방법인 것 같아서 조심스럽습니다.
- 좋은 방법에 대한 조언 부탁드립니다. 🙏🏻

WhalesJin committed:
feat: 버튼들 연결

WhalesJin committed:
feat: 버튼에 함수 넣기

WhalesJin committed:
refactor: 함수 이름 변경

takeOrder를 didTab으로

WhalesJin committed:
refactor: 쥬스 주문 버튼 함수 수정

쥬스 완성 / 재료 부족 Alert 띄우기

WhalesJin committed:
feat: 예 버튼 클릭 시 화면 전환 함수 생성

WhalesJin committed:
feat: 재고량 레이블 생성

- 과일 레이블.text에 readCurrentStock 함수로 현재 재고량 보이기
- readCurrentStock 함수의 오류 대신 0을 반환하도록 수정
- readCurrentStock 함수의 throws와 try 다 삭제

WhalesJin committed:
refactor: 얼럿의 핸들러 부분을 후행클로저로 수정

WhalesJin committed:
refactor: 바나나 쥬스 버튼 함수 기능 분리

- showCompletionAlert, showFailureAlert 함수로 쪼개기
- showAlert 함수에서 조건문으로 고르고
- 버튼 함수에서 showAlert 함수 호출 및 재고 레이블 띄우기

WhalesJin committed:
feat: 다른 과일 쥬스 버튼 함수 구현

WhalesJin committed:
refactor: 과일 재고함 은닉화

- 과일 하나의 재고를 불러오는 함수 삭제
- 과일 재고함 불러오는 함수 생성
- 연관된 함수들 수정

WhalesJin committed:
refactor: 쥬스메이커 안의 과일가게 프로퍼티 은닉화

- 과일 가게의 과일 재고함 불러오는 함수 생성
- 프린트문 주석들 제거

WhalesJin committed:
refactor: 뷰 컨트롤러의 변수들 은닉화

WhalesJin committed:
refactor: 함수명 오타 수정

WhalesJin committed:
feat: 레이블에 현재 재고량 넣어주는 함수 생성

WhalesJin committed:
docs: README.md 수정

WhalesJin committed:
docs: README.md 수정

WhalesJin committed:
feat: 주문 버튼 함수 생성

과일 주문 버튼 함수들 통합

JoSH0318 commented on JuiceMaker/JuiceMaker/Controller/ViewController.swift:
```diff
@@ -8,11 +8,108 @@ import UIKit
+    
     override func viewDidLoad() {
         super.viewDidLoad()
         // Do any additional setup after loading the view.
```
필요없는 주석은 없어도 될 것 같아요! 😀

WhalesJin replied:
넵넵 지웠습니다!!😃

JoSH0318 commented on JuiceMaker/JuiceMaker/Controller/ViewController.swift:
```diff
@@ -8,11 +8,108 @@ import UIKit
 
 class ViewController: UIViewController {
```
ViewController 네이밍을 해주면 좋겠어요!(파일명도 수정해야겠어요 !)

WhalesJin replied:
해당하는 뷰컨트롤러의 뷰에서는 쥬스 주문을 해서
`JuiceOrderViewController`로 네이밍을 해줬습니다!

JoSH0318 commented on JuiceMaker/JuiceMaker/Controller/ViewController.swift:
```diff
@@ -8,11 +8,108 @@ import UIKit
+    @IBOutlet private weak var kiwiStockLabel: UILabel!
+    @IBOutlet private weak var mangoStockLabel: UILabel!
+    
+    private let juiceMaker = JuiceMaker()
```
1. 반복되는 프로퍼티의 성격별로 개행을 주신 걸까요? 너무 좋습니다 👍
2. weak은 어떤 키워드인가요?

JoSH0318 commented on JuiceMaker/JuiceMaker/Controller/ViewController.swift:
```diff
@@ -8,11 +8,108 @@ import UIKit
+    }
+    
+    @IBAction func didTapStockChangeButton(_ sender: UIBarButtonItem) {
+        if let pushStockChangeViewController = self.storyboard?.instantiateViewController(withIdentifier: "stockChange") {
+            self.navigationController?.pushViewController(pushStockChangeViewController,
+                                                          animated: true)
+        }
```
if문을 활용한 이유가 있을까요? guard let을 사용하는 것에 대해서는 어떻게 생각하시나요? 둘의 차이점은 무엇인가요?

WhalesJin replied:
표면적으로 가장 큰 차이는
`guard 문`은 항상 `else`랑 같이 구현되면서
조건이 `false`인 경우를 먼저 명시해줘서 원하는 조건이 아닌 경우에는 `early exit` 즉, 함수를 빨리 끝내게 됩니다.

반면에 `if 문`은 참일 때를 먼저 확인해주고 `else`는 필요하면 넣어주지만 생략도 가능하고, 여러 가지의 조건이 생기는 경우에는 `else if`로 '아닌 경우 중에 이런 조건, 또 아닌 경우 중에 이런 조건' 이런 식으로 아닌 경우들을 세분화 할 수 있습니다.

지금 같은 경우에는 `else`일 때의 처리가 애매하다고 생각이 되어 `if 문`을 썼었는데 반환값이 없는 `void` 타입의 함수이므로 `return`만 넣어서 `else`일 때를 처리해 줄 수 있다는 것을 배웠고, 그렇게 되면 아닌 경우 빨리 종료가 되는 `guard  문`이 좋을 것 같아서 수정했습니다.

﻿

JoSH0318 commented on JuiceMaker/JuiceMaker/Model/JuiceMaker.swift:
```diff
@@ -13,15 +13,20 @@ struct JuiceMaker {
+    func takeOrder(_ juice: Juice) -> Bool {
         do {
             try makeJuice(juice)
+            return true
         } catch FruitStoreError.invalidFruit {
-            print("올바르지 않은 과일 이름입니다.")
+            return false
         } catch FruitStoreError.insufficientFruit {
-            print("과일의 재고가 부족하여 쥬스를 만들 수 없습니다.")
+            return false
         } catch {
-            print("알 수 없는 에러입니다.")
+            return false
         }
     }
```
error handling을 이렇게 사용하게 된다면 viewController 입장에서는 false가 어떤 error 떄문에 반환된지 모를 것 같아요.
아래 코드처럼 error handling을 ViewController에서 할수 있도록 thows 하는 게 좋겠네요.
```swift
func takeOrder(_ juice: Juice) throws {
    try makeJuice(juice)
}
```

WhalesJin replied:
오!! 저는 `ViewController`에서는 에러 처리를 못 해주는 줄 알았어요!!
더 공부해보겠습니다 감사합니다!!!
조언 주신 방향으로 코드 수정했습니다 🙇🏻‍♀️

JoSH0318 commented on JuiceMaker/JuiceMaker/Controller/ViewController.swift:
```diff
@@ -8,11 +8,108 @@ import UIKit
+        configureStockLabel()
+    }
+     
+    func showAlert(of juice: Juice) {
+        if juiceMaker.takeOrder(juice) {
+            showCompletionAlert(for: juice)
+        } else {
+            showFailureAlert()
+        }
+    }
```
`takeOrder` 메서드에서 error handling을 bool 값으로 변경하면서 이러한 코드가 생겨난 것 같아요.
`showAlert` 메서드 안에서 "주문받다"라는 메서드가 호출되는 게 어색합니다. showAlert의 역할에 벗어난 것 같아요!

WhalesJin replied:
`takeOrder`함수를 수정하고 에러 처리 하는 함수 새로 구현하면서 이 부분도 수정했습니다!!
보시고 가능하시다면 조언 부탁드립니다 🙏🏻

JoSH0318 commented on JuiceMaker/JuiceMaker/Controller/ViewController.swift:
```diff
@@ -8,11 +8,108 @@ import UIKit
+        let juiceFailureAlert = UIAlertController(title: nil,
+                                      message: "재료가 모자라요. 재고를 수정할까요?",
+                                      preferredStyle: UIAlertController.Style.alert)
+        let stockChangeAction = UIAlertAction(title: "예",
+                                              style: UIAlertAction.Style.destructive)
+        { stockChangeAction in self.didTapStockChangeButton(self.stockChangeButton) }
```
해당 부분 코드 컨벤션이 어색해요!
또한, 클로저 내부 `stockChangeAction` 파라미터는 사용 중인가요? 사용하지 않는다면 어떻게 해야할까요?

WhalesJin replied:
`stockChangeAction` 파라미터는 따로 사용하지 않아서 `_`로 대체하였습니다.
컨벤션도 수정하였습니다 😊

JoSH0318 reviewed (commented):
안녕하세요 웨일 🐳

혼자 고생이 많으십니다🥹
고민의 흔적이 많이 느껴지고, 기교 없이 깔끔한 코드라고 생각이 들어요👍 너무 좋습니다.
때문에 특히 가독성 측면에서 좋은 코드라고 생각을 합니다. 
다만 이런 저런 시도를 해보면 더 재밌는 코드가 나올 것 같아요! 화이팅입니다!

몇 가지 코멘트, 답변, 질문 남겼습니다!

--- 

### 답변 

Q: 함수를 통합하는 더 좋은 방법
여러 개의 반복되는 함수를 하나로 합치는 과정을 오래 고민했습니다.
검색해보니 tag 키워드를 쓰는 방법을 찾았는데 그렇게 되면 숫자 입력이 들어가야하는데 쥬스주문에는 정해진 순서나 명확한 숫자가 정해지지 않는다고 생각해서 채택하지 않았습니다. ([참고 블로그](https://g1embed.tistory.com/8))
버튼 이름에 쥬스 타입 케이스 이름과 같은 단어가 들어있어서 그걸 가져다 쓰면 좋겠다라는 생각을 했는데 결국 방법을 찾지 못했습니다.
가장 현실적인 방법으로 switch 문을 선택하였는데 약간의 무식한 방법인 것 같아서 조심스럽습니다.
좋은 방법에 대한 조언 부탁드립니다. 🙏🏻 

> A: 지금도 좋은 방법이라고 생각합니다. 무식한 방법이라는 건 없어요! 때론 무식할 때, 성능과 가독성을 다 얻기도 하더라구요! 절대로 무식하다고 볼 수 없는거죠! 
말씀하신대로 tag를 활용하는 방법, [Initializing from a Raw Value](https://docs.swift.org/swift-book/documentation/the-swift-programming-language/enumerations/#Initializing-from-a-Raw-Value)를 활용하는 방법도 있을 것 같아요. 이것말고도 방법은 무궁무진할 거예요. 더 좋은 아이디어가 생각난다면 얼마든지 코드에 적용해주세요! 재밌을 것 같지 않나요?😆 

---

### 질문

Q: 화면전환 방법을 선택하신 고민사항을 잘 읽었어요! 하지만 그 근거가 조금은 부족하다라는 생각이 들어요! 다시 H.I.G를 참고해보시면 어떨까요? 그 근거를 찾을 수 있을 것 같아요! 꼭!!! 답변해주시면 좋을 것 같습니다!

WhalesJin committed:
refactor: 컨벤션 수정

WhalesJin committed:
refactor: 화면 전환 조건문 수정

WhalesJin committed:
feat: 뷰컨트롤러에서 주문하며 에러 처리 하는 함수 구현

- showAlert 함수 삭제
- invalidFruit 에러 삭제
- takeOrder함수는 makeJuice함수 호출만 하게 수정
- order함수가 에러 처리하고 alert 띄우도록 구현

WhalesJin committed:
refactor: 에러 처리 함수 수정

WhalesJin committed:
feat: 에러 알리는 안내창 띄우는 함수 생성

JoSH0318 commented on JuiceMaker/JuiceMaker/Model/JuiceMaker.swift:
```diff
@@ -13,16 +13,12 @@ struct JuiceMaker {
+        return fruitStore.readFruitInventory()
+    }
+    
+    func takeOrder(_ juice: Juice) throws {
+        try makeJuice(juice)
```
takeOrder의 역할이 makeJuice를 호출하는 것이 다인 것 같아요.
makeJuice의 코드를 이곳으로 옮겨와도 될 것 같습니다.

WhalesJin replied:
`makeJuice`의 은닉화가 필요하다고 생각했는데 `makeJuice` 안에서 불리는 함수들이 은닉화가 필요한거지 저 친구는 괜찮을 것 같아서 주신 말씀 참고해서 수정했습니다.
감사합니다!!🙏🏻

JoSH0318 commented on JuiceMaker/JuiceMaker/Controller/JuiceOrderViewController.swift:
```diff
@@ -0,0 +1,157 @@
+    
+    func showFailureAlert() {
+        let juiceFailureAlert = UIAlertController(
+            title: nil,
+            message: "재료가 모자라요. 재고를 수정할까요?",
+            preferredStyle: UIAlertController.Style.alert
+        )
+        
+        let stockChangeAction = UIAlertAction(
+            title: "예",
+            style: UIAlertAction.Style.destructive
+        ) { _ in
+            self.didTapStockChangeButton(self.stockChangeButton)
+        }
+        
+        let closeAction = UIAlertAction(
+            title: "아니오",
+            style: UIAlertAction.Style.cancel
+        )
+        
+        juiceFailureAlert.addAction(stockChangeAction)
+        juiceFailureAlert.addAction(closeAction)
+        present(juiceFailureAlert, animated: true, completion: nil)
+    }
+    
+    func showErrorAlert() {
+        let errorAlert = UIAlertController(
+            title: nil,
+            message: "알 수 없는 에러가 발생하였습니다. \n 다시 시도해주시기 바랍니다.",
+            preferredStyle: UIAlertController.Style.alert
+        )
+        
+        let closeAction = UIAlertAction(
+            title: "닫기",
+            style: UIAlertAction.Style.default
+        )
+        
+        errorAlert.addAction(closeAction)
+        present(errorAlert, animated: true, completion: nil)
+    }
```
1. 일관된 컨벤션 때문에 가독성이 높은 코드네요. 👍
2. 반면에 반복되는 코드가 많습니다. 조금이나마 줄여볼 수 있을 것 같아요. 시도해보시면 좋을 것 같습니다.

WhalesJin replied:
1. 칭찬 감사드립니닷 (아주아주 기분이 좋습니다 ๑❛ᴗ❛๑ )
2. 도전해보겠습니다!!!!🙋🏻‍♀️!!!

WhalesJin committed:
refactor: 변경된 재고 띄우는 함수 호출 위치 변경

JoSH0318 commented on JuiceMaker/JuiceMaker/Controller/JuiceOrderViewController.swift:
```diff
@@ -0,0 +1,157 @@
+    
+    @IBAction func didTapStockChangeButton(_ sender: UIBarButtonItem) {
+        guard let pushStockChangeViewController = self.storyboard?.instantiateViewController(
+            withIdentifier: "stockChange"
```
identifier는 일반적으로 객체명을 사용해요! 이점 참고하시면 좋을 것 같습니다.

WhalesJin replied:
재고수정 `view`의 `ViewController`가 객체가 될거라
`stockChangeViewController` 로 이름을 변경했습니다!!

JoSH0318 commented on JuiceMaker/JuiceMaker/Controller/JuiceOrderViewController.swift:
```diff
@@ -0,0 +1,157 @@
+        }
+        
+        order(juice)
+        configureStockLabel()
```
메서드 호출 시점을 다시 한번 생각해보시면 좋을 것 같아요!

WhalesJin replied:
쥬스 완성 얼럿이 뜨기 직전에 재고가 변경되어야 한다고 생각해서
메서드 호출 시점을 쥬스 완성 얼럿 직전으로 바꿔주었습니다!!

JoSH0318 reviewed (commented):
안녕하세요 웨일!

몇 가지 코멘트 남겼습니다!
점점 더 성장하는 모습이 보이는 것 같아요!
조금만 더 힘냅시다! 👍💪

WhalesJin committed:
refactor: makeJuice함수를 takeOrder함수로 변경

WhalesJin committed:
fix: takeOrder함수 은닉화 해제

WhalesJin committed:
refactor: 화면전환방법 모달로 변경

WhalesJin commented:
주신 질문에 답을 드려봅니다.

> Q: 화면전환 방법을 선택하신 고민사항을 잘 읽었어요! 하지만 그 근거가 조금은 부족하다라는 생각이 들어요! 다시 H.I.G를 참고해보시면 어떨까요? 그 근거를 찾을 수 있을 것 같아요! 꼭!!! 답변해주시면 좋을 것 같습니다!

용도 면에서의 차이를 보면
내비게이션은 하나의 정보 흐름을 가지고 깊게 파고드는 형식으로, 정보의 깊이와 흐름을 가지는 구조에서 사용되고
모달은 사용자의 이목을 끌기 위해 사용되며 되도록 단순하고 사용자가 빠르게 처리할 수 있는 내용을 표현하는 것이 좋다고 합니다.

쥬스메이커의 재고 수정 화면은 정보의 흐름보다는 단순하게 재고수량을 조정하는 화면이기 때문에 모달이 더 적절할 거라 생각해서 모달로 수정하였습니다.

WhalesJin committed:
feat: Alert 열거형 타입 생성

WhalesJin committed:
feat: Alert 띄우는 함수 생성

showCompletion/Failure/ErrorAlert 함수 삭제

JoSH0318 commented on JuiceMaker/JuiceMaker/Controller/ResultAlert.swift:
```diff
@@ -0,0 +1,36 @@
+
+import UIKit
+
+enum ResultAlert {
```
좋은 네이밍이라고 생각이 들어요! 이런 네이밍을 생각 못해봤는데 좋습니다!
하지만 조금 포괄적인 의미를 담은 네이밍인 것 같아요!
결과 알럿의 종류를 담았는데 조금 더 디테일하면 좋겠어요! 
어떤 결과 알럿인가요? 이것을 생각하면 좋을 것 같아요!

WhalesJin replied:
쥬스 주문의 결과이고, 얼럿을 위해서 만든 열거형이긴하지만
자체로는 얼럿의 특성을 담고 있지 않아서 `JuiceOrderResult`로 이름 수정했습니다!!

JoSH0318 commented on JuiceMaker/JuiceMaker/Controller/JuiceOrderViewController.swift:
```diff
@@ -0,0 +1,137 @@
+                message: resultAlert.message,
+                preferredStyle: UIAlertController.Style.alert
+            )
+        }
```
`message`에 juice name을 넣기 위해 분기처리 해준것 같아요!
이런 경우 구현하신 ResultAlert에 연관값(Associated Values)을 이용하면 좋을 것 같아요!

WhalesJin replied:
알려주신 방법 참고해서 연관값으로 다시 정리해보았어요!!
맘에 들지 않았던 코드를 바꿀 수 있어서 너무 좋습니다🤩
감사합니다!!

WhalesJin committed:
refactor: completion 케이스에 연관값으로 쥬스 넣기

juice 이름 넣기 위한 조건문 삭제

WhalesJin committed:
refactor: JuiceOrderResult 열거형 이름 수정

JoSH0318 commented on JuiceMaker/JuiceMaker/Controller/JuiceOrderResult.swift:
```diff
@@ -0,0 +1,36 @@
+
+import UIKit
+
+enum JuiceOrderResult: Equatable {
```
👍

JoSH0318 reviewed (commented):
고생하셨습니다. 웨일 😀

요구사항 만족하셨고, 특별히 언급할 점이 없어 merge 하도록 하겠습니다.
다음 step도 화이팅입니다!
